kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/KuraException.java

```diff
@@ -20,167 +18,191 @@
-	 * 
-	 * @param cause
-	 * @param message
-	 * @return
-	 */
-	public static KuraException internalError(String message) {
-		return new KuraException(KuraErrorCode.INTERNAL_ERROR, null, message);
-	}
-
-	
-	public KuraErrorCode getCode() {
-		return m_code;
-	}
-
-	
-	public String getMessage() {
-		return getLocalizedMessage(Locale.US);
-	}
-
-	
-	public String getLocalizedMessage() {
-		return getLocalizedMessage(Locale.getDefault());
-	}
-
-	
-	private String getLocalizedMessage(Locale locale) 
-	{
-		String pattern = getMessagePattern(locale, m_code);
-		if (m_code == null || KuraErrorCode.INTERNAL_ERROR.equals(m_code)) {
-			if (m_arguments != null && m_arguments.length > 1) {
-				// append all arguments into a single one			
-				StringBuilder sbAllArgs = new StringBuilder();
-				for (Object arg : m_arguments) {
-					sbAllArgs.append(" - ");
-					sbAllArgs.append(arg);
-				}
-				m_arguments = new Object[] {sbAllArgs.toString()};
-			}
-		}
-		String message = MessageFormat.format(pattern, m_arguments);
-		return message;
-	}
-	
-
-	private String getMessagePattern(Locale locale, KuraErrorCode code) 
-	{
-		//
-		// Load the message pattern from the bundle
-		String messagePattern = null;
-		ResourceBundle resourceBundle = null;
-		try {
-			
-			resourceBundle = ResourceBundle.getBundle(KURA_EXCEPTION_MESSAGES_BUNDLE, locale);
-			if (resourceBundle != null && code != null) {
-				messagePattern = resourceBundle.getString(code.name());
-				if (messagePattern == null) {
-					s_logger.warn("Could not find Exception Messages for Locale {} and code {}", locale, code);
-				}
-			}
-		} catch (MissingResourceException mre) {
-			// log the failure to load a message bundle
-			s_logger.warn("Could not load Exception Messages Bundle for Locale {}", locale);
-		}
-
-		//
-		// If no bundle or code in the bundle is found, use a generic message
-		if (messagePattern == null) {
-			if (code != null) {
-				// build a generic message format
-				messagePattern = MessageFormat.format(KURA_GENERIC_MESSAGES_PATTERN, code.name());
-			}
-			else {
-				// build a generic message format
-				messagePattern = MessageFormat.format(KURA_GENERIC_MESSAGES_PATTERN, "Unknown");			
-			}
-		}
-
-		return messagePattern;
-	}
+public class KuraException extends Exception {
+
+    /** The Constant denoting resource bundle. */
+    private static final String KURA_EXCEPTION_MESSAGES_BUNDLE = "org.eclipse.kura.core.messages.KuraExceptionMessagesBundle";
+
+    /** The Constant denoting message pattern. */
+    private static final String KURA_GENERIC_MESSAGES_PATTERN = "Generic Error - {0}: {1}";
+
+    /** The Logger. */
+    private static final Logger s_logger = LoggerFactory.getLogger(KuraException.class);
+
+    /** The Constant denoting serial version identifier. */
+    private static final long serialVersionUID = 7468633737373095296L;
+
+    /** The associated arguments. */
+    private transient Object[] arguments;
```

Comment:
Why are the arguments `transient`? I think we only should allow `Serializable` in general.